### PR TITLE
SONAR-6355 Period 1 is now leak period

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by simonbrandhof** [comment]
In general settings, description of properties sonar.timemachine.period2/period3 should not be "See the property 'Period 1'" but "See the property 'Leak Period'"

**Comment by simonbrandhof** [comment]
Default Quality gate "SonarQube way" is not correct. Thresholds relate to period "over 30 days".

**Comment by julienlancelot** [comment]
Oups...

**Comment by julienlancelot** [comment]
Thanks for the feedbacks, I've applied them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java [hunk 1]
```diff
@@ -20,15 +20,25 @@
...
+import static org.sonar.api.measures.CoreMetrics.NEW_COVERAGE_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_CRITICAL_VIOLATIONS_KEY;
+import static org.sonar.api.measures.CoreMetrics.NEW_SQALE_DEBT_RATIO_KEY;
+import static org.sonar.db.qualitygate.QualityGateConditionDto.OPERATOR_GREATER_THAN;
+import static org.sonar.db.qualitygate.QualityGateConditionDto.OPERATOR_LESS_THAN;
+
 public class RegisterQualityGates implements Startable {
 
   private static final String BUILTIN_QUALITY_GATE = "SonarQube way";
+  private static final int PERIOD = 1;
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java]: To be renamed `WATER_LEAK_PERIOD` ?

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 2]
```diff
@@ -323,9 +323,9 @@
   String TIMEMACHINE_MODE_VERSION = "version";
   String TIMEMACHINE_MODE_DAYS = "days";
   String TIMEMACHINE_MODE_PREVIOUS_VERSION = "previous_version";
-  String TIMEMACHINE_DEFAULT_PERIOD_1 = TIMEMACHINE_MODE_PREVIOUS_ANALYSIS;
-  String TIMEMACHINE_DEFAULT_PERIOD_2 = "30";
-  String TIMEMACHINE_DEFAULT_PERIOD_3 = TIMEMACHINE_MODE_PREVIOUS_VERSION;
+  String TIMEMACHINE_DEFAULT_PERIOD_1 = TIMEMACHINE_MODE_PREVIOUS_VERSION;
+  String TIMEMACHINE_DEFAULT_PERIOD_2 = TIMEMACHINE_MODE_PREVIOUS_ANALYSIS;
+  String TIMEMACHINE_DEFAULT_PERIOD_3 = "30";
```
Comment (simonbrandhof) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Changing these constants do not work if plugins use a previous version of API. Indeed constant values are inlined by compiler. I suggest to move them from API to core.

**Comment by simonbrandhof** [comment]
LGTM

File: server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java]: ok

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**